conductor/src/main/java/com/bluelinelabs/conductor/internal/OwnViewTreeLifecycleAndRegistry.kt

```diff
@@ -0,0 +1,118 @@
+            }
+
+            override fun postCreateView(controller: Controller, view: View) {
+                ViewTreeLifecycleOwner.set(view, this@OwnViewTreeLifecycleAndRegistry)
+                ViewTreeSavedStateRegistryOwner.set(
+                    view,
+                    this@OwnViewTreeLifecycleAndRegistry
+                )
+                lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_START)
+            }
+
+            override fun postAttach(controller: Controller, view: View) {
+                lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
+            }
+
+            override fun preDetach(controller: Controller, view: View) {
+                lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+            }
+
+            override fun onSaveInstanceState(controller: Controller, outState: Bundle) {
+                savedStateRegistryController.performSave(outState)
+            }
+
+            override fun onSaveViewState(controller: Controller, outState: Bundle) {
+                savedRegistryState = Bundle()
+                savedStateRegistryController.performSave(savedRegistryState)
+            }
+
+            override fun onRestoreInstanceState(
+                controller: Controller,
+                savedInstanceState: Bundle
+            ) {
+                initLifecycle()
+                savedStateRegistryController.performRestore(savedInstanceState)
+                hasRestoredInstanceState = true
+            }
+
+            override fun preDestroyView(controller: Controller, view: View) {
+                lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+                if (controller.isBeingDestroyed && controller.router.backstackSize == 0) {
+                    val parent = view.parent as? View
+                    handlingDestroyViaHostDetach = parent != null
+                    parent?.addOnAttachStateChangeListener(object :
+                        View.OnAttachStateChangeListener {
+                        override fun onViewAttachedToWindow(v: View?) = Unit
+                        override fun onViewDetachedFromWindow(v: View?) {
+                            parent.removeOnAttachStateChangeListener(this)
+                            onDestroy()
+                        }
+                    })
+                }
+            }
+
+            override fun preDestroy(controller: Controller) {
+                if (!handlingDestroyViaHostDetach) {
+                    onDestroy()
+                }
+            }
+
+            private fun onDestroy() {
+                lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+            }
+        })
+    }
+
+    override fun getLifecycle(): Lifecycle = lifecycleRegistry
+
+    override fun getSavedStateRegistry(): SavedStateRegistry =
+        savedStateRegistryController.savedStateRegistry
+
+    companion object {
+        fun own(target: Controller) {
```

Comment:
Would be nice to have a comment explaining what/why this is

Reply:
What do you mean? What the class is for? It's not pubic api (at least not on kotlin) 😉

Reply:
Yeah, what the class is for. It may not be public, but I have _somewhat_ of an understanding of Conductor's internals, and it isn't clear to me what it's achieving. Feel like it might be hard to remember context a year from now.

Reply:
I'm a Conductor user who just tried naively using a `ComposeView` in my `Controller` and hit lifecycle related exceptions.  Searching led me here; but this PR, nor anything else I can find, leaves me with a clue how to actually make Compose work with Conductor...

Reply:
> This class sets the [ViewTreeLifecycleOwner] and [ViewTreeSavedStateRegistryOwner] which is necessary for Jetpack Compose. By setting these
How!??
> the view state restoration and compose lifecycle play together with the lifecycle of the [Controller].
Example, please.

Reply:
Can you open a bug report with a complete stack trace and the conductor & compose versions you are using?